### PR TITLE
Add support for parsing Wilderness assignment updates to slayer plugin

### PR DESCRIPTION
**Pull request by robinwithes**
Fixes https://github.com/runelite/runelite/issues/5687

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 1]
```diff
@@ -78,7 +78,7 @@
 public class SlayerPlugin extends Plugin
 {
 	//Chat messages
-	private static final Pattern CHAT_GEM_PROGRESS_MESSAGE = Pattern.compile("You're assigned to kill (.*); only (\\d*) more to go\\.");
+	private static final Pattern CHAT_GEM_PROGRESS_MESSAGE = Pattern.compile("You're assigned to kill (.*?)(in the Wilderness)?; only (\\d*) more to go\\.");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: just add space before in instead doing what are you doing at line 427

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 2]
```diff
@@ -418,13 +418,13 @@ public void onChatMessage(ChatMessage event)
...
 		Matcher mProgress = CHAT_GEM_PROGRESS_MESSAGE.matcher(chatMsg);
+
 		if (!mProgress.find())
 		{
 			return;
 		}
 		String taskName = mProgress.group(1);
-		int amount = Integer.parseInt(mProgress.group(2));
-
+		int amount = Integer.parseInt(mProgress.group(3));
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Don't change this. Make the "in the Wilderness" group a non-capturing group in the regex.

Comment (robinwithes) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Im really shit with regex, how do i do this?

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: https://www.regular-expressions.info/brackets.html#noncap

```
private static final Pattern CHAT_GEM_PROGRESS_MESSAGE = Pattern.compile("You're assigned to kill (.*?)(?:in the Wilderness)?; only (\\d*) more to go\\.");
```

**Comment by Nightfirecat** [comment]
Also, there should be a test added for this case.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java (hunk 2 above)
Comment (robinwithes) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Thanks!

**Comment by robinwithes** [comment]
> 
> 
> Also, there should be a test added for this case.

You mean a video showing the fix actually works or something?

File: runelite-client/src/test/java/net/runelite/client/plugins/slayer/SlayerPluginTest.java [hunk 3]
```diff
@@ -55,9 +58,12 @@
 {
 	private static final String TASK_NEW = "Your new task is to kill 231 Suqahs.";
 	private static final String TASK_NEW_NPC_CONTACT = "Excellent, you're doing great. Your new task is to kill<br>211 Suqahs.";
+	private static final String TASK_CHECKSLAYERGEM_WILDERNESS = "You're assigned to kill Suqahs in the Wilderness; only 211 more to go.";
+	private static final String TASK_CHECKSLAYERGEM = "You're assigned to kill Suqahs; only 211 more to go.";
 
 	private static final String TASK_BOSS_NEW = "Excellent. You're now assigned to kill Vet'ion 3 times.<br>Your reward point tally is 914.";
 
+	private static final Pattern CHAT_GEM_PROGRESS_MESSAGE = Pattern.compile("You're assigned to kill (.*?)(?: in the Wilderness)?; only (\\d*) more to go\\.");
```
Comment (deathbeam) [inline runelite-client/src/test/java/net/runelite/client/plugins/slayer/SlayerPluginTest.java]: This looks unused why is it here

Comment (robinwithes) [reply runelite-client/src/test/java/net/runelite/client/plugins/slayer/SlayerPluginTest.java]: Ah sorry, it was from the first test i made.

**Pull request merged**